pkg/transforms/mqttsecret_test.go

```diff
@@ -0,0 +1,388 @@
+	})
+
+	require.Nil(t, err)
+	assert.NotNil(t, sender.opts.TLSConfig.ClientCAs)
+	assert.Empty(t, sender.opts.Username)
+	assert.Empty(t, sender.opts.Password)
+	assert.Nil(t, sender.opts.TLSConfig.Certificates)
+}
+func TestConfigureMQTTClientWithClientCert(t *testing.T) {
+	sender := NewMQTTSecretSender(MQTTSecretConfig{}, false)
+	sender.mqttConfig = MQTTSecretConfig{
+		AuthMode: AuthModeCert,
+	}
+	err := sender.configureMQTTClientForAuth(mqttSecrets{
+		username:     "username",
+		password:     "password",
+		certpemblock: []byte(testclientcert),
+		keypemblock:  []byte(testclientkey),
+		capemblock:   []byte(testcacert),
+	})
+	require.Nil(t, err)
+	assert.Empty(t, sender.opts.Username)
+	assert.Empty(t, sender.opts.Password)
+	assert.NotNil(t, sender.opts.TLSConfig.Certificates)
+	assert.NotNil(t, sender.opts.TLSConfig.ClientCAs)
+}
+
+// should this fail?
+func TestConfigureMQTTClientWithClientCertNoCA(t *testing.T) {
+	sender := NewMQTTSecretSender(MQTTSecretConfig{}, false)
+	sender.mqttConfig = MQTTSecretConfig{
+		AuthMode: AuthModeCert,
+	}
+	err := sender.configureMQTTClientForAuth(mqttSecrets{
+		username:     "username",
+		password:     "password",
+		certpemblock: []byte(testclientcert),
+		keypemblock:  []byte(testclientkey),
+	})
+
+	require.Nil(t, err)
+	assert.Empty(t, sender.opts.Username)
+	assert.Empty(t, sender.opts.Password)
+	assert.NotNil(t, sender.opts.TLSConfig.Certificates)
+	assert.Nil(t, sender.opts.TLSConfig.ClientCAs)
+}
+func TestConfigureMQTTClientWithNone(t *testing.T) {
+	sender := NewMQTTSecretSender(MQTTSecretConfig{}, false)
+	sender.mqttConfig = MQTTSecretConfig{
+		AuthMode: AuthModeNone,
+	}
+	err := sender.configureMQTTClientForAuth(mqttSecrets{})
+
+	require.Nil(t, err)
+}
+
+func TestSetRetryDataPersistFalse(t *testing.T) {
+	sender := NewMQTTSecretSender(MQTTSecretConfig{}, false)
+	sender.mqttConfig = MQTTSecretConfig{}
+	sender.setRetryData(context, []byte("data"))
+	assert.Nil(t, context.RetryData)
+}
+func TestSetRetryDataPersistTrue(t *testing.T) {
+	sender := NewMQTTSecretSender(MQTTSecretConfig{}, true)
+	sender.mqttConfig = MQTTSecretConfig{}
+	sender.setRetryData(context, []byte("data"))
+	assert.Equal(t, []byte("data"), context.RetryData)
+}
+
+func TestMQTTSendNoParams(t *testing.T) {
+	sender := NewMQTTSecretSender(MQTTSecretConfig{}, true)
+	sender.mqttConfig = MQTTSecretConfig{}
+	continuePipeline, result := sender.MQTTSend(context)
+	require.False(t, continuePipeline)
+	require.Error(t, result.(error))
+}
+
+// REQUIRES BROKER
+func TestMQTTSendWithData(t *testing.T) {
+	sender := NewMQTTSecretSender(MQTTSecretConfig{}, true)
+	sender.mqttConfig = MQTTSecretConfig{
+		SecretPath: "/mqtt",
+	}
+	sender.client = MQTT.NewClient(sender.opts)
+	mockSecretProvider := security.NewSecretProvider(nil, nil)
+	mockSecretProvider.SecretClient = &mockMQTTSecretClient{}
+	context.SecretProvider = mockSecretProvider
+	sender.MQTTSend(context, "sendme")
+	// require.True(t, continuePipeline)
```

Comment:
Should these be uncommitted? and set continuePipeline & result?
Is this expected to fail?

Reply:
it will fail if dont have a broker, wont if you do /shrug

Reply:
Move to file that is for when have broker? Nothing being validate otherwise.